src/plugin/datasource.ts

```diff
@@ -515,4 +575,51 @@ export function WavefrontDatasource(instanceSettings, $q, backendSrv, templateSr
         return this.backendSrv.datasourceRequest(reqConfig);
     };
 
+
+    function refreshToken() {
+        if (instanceSettings.jsonData.cspAPIToken) {
```

Comment:
This code is same as what we are doing at line#35 and 51.
Can we create two functions like: fetchAccessTokenUsingCSPAPIToken and fetchAccessTokenUsingCSPOAuth and use these functions during initialisation and interval update